session/session.go

```diff
@@ -2236,6 +2237,49 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	return recordSet, nil
 }
 
+func copyAndPrintRecordSet(ctx context.Context, rs sqlexec.RecordSet, vars *variable.SessionVars,
+	sql string) (sqlexec.RecordSet, error) {
+	fieldTypes := make([]*types.FieldType, len(rs.Fields()))
+	for i, field := range rs.Fields() {
+		fieldTypes[i] = &field.Column.FieldType
+	}
+
+	chk := rs.NewChunk(nil)
+	var records []chunk.Row
+	var resultRows []string
+	for {
+		if err := rs.Next(ctx, chk); err != nil {
+			return nil, err
+		}
+		if chk.NumRows() == 0 {
+			break
+		}
+		// Copy and collect rows
+		for i := 0; i < chk.NumRows(); i++ {
+			row := chk.GetRow(i)
+			records = append(records, row)
+			resultRows = append(resultRows, row.ToString(fieldTypes))
+		}
+		chk = chunk.Renew(chk, 32)
+	}
+
+	// Print all rows
+	logutil.Logger(ctx).Info("statement returned result",
```

Comment:
If there are quite a lot of rows, there could be memory risk.

Reply:
Can we assume that the transaction sizes are not too big in the environment that we'are going to use this?
Another way to print them is to collect them when the `resultSet` is being consumed by calling `Next`